src/app/projects/projects.component.ts

```diff
@@ -114,20 +114,42 @@ export class ProjectsComponent implements OnInit, OnDestroy {
             .finally(() => {
               this.snackBarRef.dismiss();
               this.isLoading = false;
-              console.log('got', this.allApps.length, 'apps in', (new Date()).getTime() - start, 'ms');
             })
-            .subscribe(projects => {
-              this.allApps = _.concat(this.allApps, projects);
-              // filter component gets all apps
-              this.filterApps = this.allApps;
+            .subscribe((projects: Project[]) => {
+              // Get all shapefiles.
+              this.documentService.getAll("SHAPEFILE")
+              .takeUntil(this.ngUnsubscribe)
+              .subscribe(shapefiles => {
+                  // "Attach" a shapefile to its project.
+                  if (shapefiles) {
+                    projects.forEach(project => {
+                      // project.shapefiles = [];
+                      shapefiles.forEach(shapefile => {
+                        if (project._id === shapefile.project) {
+                          console.log("how many times")
```

Comment:
another console.log left in, guessing, not suppose to be here?

Reply:
It's not, sorry about that. Will update with them removed